mmpose/datasets/transforms/common_transforms.py

```diff
@@ -873,58 +874,136 @@ class GenerateTarget(BaseTransform):
-        self.target_type = target_type
+        self.multilevel = multilevel
         self.use_dataset_keypoint_weights = use_dataset_keypoint_weights
 
-        if self.target_type == 'multilevel_heatmap':
-            if not isinstance(self.encoder_cfg, list):
-                raise ValueError(
-                    'The encoder should be a list if target type is '
-                    '"multilevel_heatmap"')
+        if isinstance(self.encoder_cfg, list):
             self.encoder = [
                 KEYPOINT_CODECS.build(cfg) for cfg in self.encoder_cfg
             ]
         else:
+            assert not self.multilevel, (
+                'Need multiple encoder configs if ``multilevel==True``')
             self.encoder = KEYPOINT_CODECS.build(self.encoder_cfg)
 
     def transform(self, results: Dict) -> Optional[dict]:
+        """The transform function of :class:`GenerateTarget`.
+
+        See ``transform()`` method of :class:`BaseTransform` for details.
+        """
+
+        if results.get('transformed_keypoints', None) is not None:
+            # use keypoints transformed by TopdownAffine
+            keypoints = results['transformed_keypoints']
+        elif results.get('keypoints', None) is not None:
+            # use original keypoints
+            keypoints = results['keypoints']
+        else:
+            raise ValueError(
+                'GenerateTarget requires \'transformed_keypoints\' or'
+                ' \'keypoints\' in the results.')
+
+        keypoints_visible = results['keypoints_visible']
+
+        # Encoded items from the encoder(s) will be updated into the results.
+        # Please refer to the document of the specific codec for details about
+        # encoded items.
+        if not isinstance(self.encoder, list):
+            # For single encoding, the encoded items will be directly added
+            # into results.
+            encoded = self.encoder.encode(
+                keypoints=keypoints, keypoints_visible=keypoints_visible)
+
+        else:
+            encoded_list = [
+                _encoder.encode(
+                    keypoints=keypoints, keypoints_visible=keypoints_visible)
+                for _encoder in self.encoder
+            ]
+
+            if self.multilevel:
+                # For multilevel encoding, the encoded items from each encoder
+                # should have the same keys.
+
+                keys = encoded_list[0].keys()
+                if not all(_encoded.keys() == keys
+                           for _encoded in encoded_list):
+                    raise ValueError(
+                        'Encoded items from all encoders must have the same '
+                        'keys if ``multilevel==True``.')
+
+                encoded = {
+                    k: [_encoded[k] for _encoded in encoded_list]
+                    for k in keys
+                }
+
+            else:
+                # For combined encoding, the encoded items from different
```

Comment:
Is it necessary to handle the case that we want to use different encoded `keypoint_weights`  to calculate different losses for different codecs? Should we keep all the `keypoint_weights` for possible usage?